pennylane_lightning/core/src/simulators/lightning_qubit/algorithms/AdjointJacobianLQubit.hpp

```diff
@@ -256,12 +256,8 @@ class AdjointJacobian final
         // Pointer to data storage for StateVectorLQubitRaw<PrecisionT>:
         std::unique_ptr<std::vector<std::vector<ComplexT>>> H_lambda_storage;
         size_t lambda_qubits = lambda.getNumQubits();
-        if constexpr (std::is_same_v<StateVectorLQubitManaged<PrecisionT>,
+        if constexpr (std::is_same_v<StateVectorLQubitRaw<PrecisionT>,
```

Comment:
There is a reason for these ``if constexpr`` to be as they were. 
We cannot guarantee that a catch-all ``else`` will really cover all cases.
With these changes, if a developer provides a StateVector without internal storage (like Raw) we'll have a code-wise unexpected behaviour.

Reply:
Thanks @AmintorDusko!